File: farebot-transit/src/main/java/com/codebutler/farebot/transit/TransitFactory.java
package com.codebutler.farebot.transit;

import androidx.annotation.NonNull;

import com.codebutler.farebot.card.Card;

public interface TransitFactory<C extends Card, T extends TransitInfo> {

    boolean check(@NonNull C card);

    @NonNull
    TransitIdentity parseIdentity(@NonNull C card);

    @NonNull
    T parseInfo(@NonNull C card);
}

File: farebot-transit/src/commonMain/kotlin/com/codebutler/farebot/transit/TransitFactory.kt
package com.codebutler.farebot.transit

import com.codebutler.farebot.card.Card

interface TransitFactory<C : Card, T : TransitInfo> {
    val allCards: List<CardInfo>

    fun check(card: C): Boolean

    fun parseIdentity(card: C): TransitIdentity

    fun parseInfo(card: C): T
}
